crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1263,8 +1273,11 @@ impl<RpcMiddleware> RpcServerConfig<RpcMiddleware> {
 
     /// Returns a [`CompressionLayer`] that adds compression support (gzip, deflate, brotli, zstd)
     /// based on the client's `Accept-Encoding` header
-    fn maybe_compression_layer() -> Option<CompressionLayer> {
-        Some(CompressionLayer::new())
+    fn maybe_compression_layer(disable_compression: bool) -> Option<CompressionLayer> {
+        match disable_compression {
+            true => None,
+            false => Some(CompressionLayer::new()),
```

Comment:
we can refactor this slightly to `self.disable_compression.not().then(||CompressionLayer::new())`

but a simple if else is also okay here, perhaps even better

Reply:
Sounds great. I've already refactor this using one simple if else😂.